common/txmgr/broadcaster.go

```diff
@@ -568,6 +568,12 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) hand
 		// gas limit, we could have smaller transactions "above" it that could
 		// theoretically be sent, but will instead be blocked.
 		eb.SvcErrBuffer.Append(err)
+		if attempt.TxType != 0x2 {
```

Comment:
`eb.tryAgainWithNewEstimation` will reject type 2 tx and return error, so adding a type check here